src/components/utils/src/file_system.cc

```diff
@@ -47,80 +47,116 @@
 #include <cstdio>
 #include <fstream>
 
-CREATE_LOGGERPTR_GLOBAL(logger_, "Utils")
+CREATE_LOGGERPTR_GLOBAL(logger_, "Utils::FileSystem")
 
 // Easier reference
 namespace fs = boost::filesystem;
 using boost::system::error_code;
 
 uint64_t file_system::GetAvailableDiskSpace(const std::string& path) {
+  LOG4CXX_AUTO_TRACE(logger_);
   error_code ec;
-  fs::space_info si = fs::space(path, ec);
+  fs::space_info si = {0, 0, 0};
+  si = fs::space(path, ec);
 
   if (ec) {
     // If something went wrong, assume no free space
-    return 0;
-  } else {
-    return si.free;
+    LOG4CXX_ERROR_WITH_ERRNO(logger_,
+                             "Unable to get available disk space: '"
+                                 << path << "', reason: " << ec.message());
   }
+  return si.free;
 }
 
 uint64_t file_system::FileSize(const std::string& path) {
+  LOG4CXX_AUTO_TRACE(logger_);
   error_code ec;
   // Boost returns sizes as unsigned
-  uint64_t fsize = (uint64_t)fs::file_size(path, ec);
+  const uint64_t fsize = static_cast<uint64_t>(fs::file_size(path, ec));
 
   if (ec) {
-    LOG4CXX_WARN_WITH_ERRNO(logger_, "Could not get file size: " << path);
+    LOG4CXX_ERROR_WITH_ERRNO(
+        logger_,
+        "Unable to get file size: '" << path << "', reason: " << ec.message());
     return 0;
   }
   return fsize;
 }
 
 size_t file_system::DirectorySize(const std::string& path) {
+  LOG4CXX_AUTO_TRACE(logger_);
   size_t dir_size = 0;
   error_code ec;
   // Recursively iterate through directory to accumulate file sizes
   fs::recursive_directory_iterator iter(path, ec);
-  // Directory does not exist
   if (ec) {
+    LOG4CXX_ERROR_WITH_ERRNO(logger_,
+                             "Unable to get directory size: '"
+                                 << path << "', reason: " << ec.message());
     return 0;
   }
+
   // default constructor gives end iterator
   fs::recursive_directory_iterator end;
-  while (iter != end) {
-    size_t fsize = fs::file_size(iter->path(), ec);
-    // No error means we can add the file
-    if (!ec) {
-      dir_size += fsize;
+  while (end != iter) {
+    const bool is_directory = fs::is_directory(iter->path(), ec);
+    if (ec) {
+      LOG4CXX_WARN_WITH_ERRNO(
+          logger_,
+          "Failed check if '" << iter->path()
+                              << "' is directory, reason: " << ec.message());
+    }
+
+    if (!is_directory && !ec) {
+      const size_t fsize = fs::file_size(iter->path(), ec);
+      if (ec) {
+        LOG4CXX_WARN_WITH_ERRNO(logger_,
+                                "Failed to get file_size: '"
+                                    << path << "', reason: " << ec.message());
+      } else {
+        // No error means we can add the file
+        dir_size += fsize;
+        LOG4CXX_DEBUG(logger_, "Adding: " << fsize << ", total: " << dir_size);
+      }
+    }
```

Comment:
Shouldn't this be called recursively if this entry is a directory?

Reply:
@jacobkeeler No as we have used special iterator type from boost library: `recursive_directory_iterator`
recursive_directory_iterator provides standard library compliant iteration over the contents of a directory, including recursion into its sub-directories. 
Reference: [Recursive iterator](https://www.boost.org/doc/libs/1_66_0/libs/filesystem/doc/reference.html#Class-recursive_directory_iterator)